### ISSUE TITLE
Viber menu in system tray is empty

### ISSUE THREAD

**Issue by vushe** [linuxmint/cinnamon#11294]
```
 * Cinnamon version 5.412
 * Distribution Mint 21
 * 64 bit
 * Viber 18.2.02
 ```

### Issue
Viber menu in system tray is empty. Basically, you cannot exit Viber other way than to kill it. 

### Steps to reproduce
Download Viber https://www.viber.com/en/download/ and install it (either deb package or AppImage). Run it, right-click viber icon in tray. 

![image](https://user-images.githubusercontent.com/15868217/202873354-7023606b-d853-4f2e-b435-26d4d4b04386.png)

### Expected behaviour
Menu items should be displayed

### Other information
Related to following thread: https://forums.linuxmint.com/viewtopic.php?t=379658

**Comment by mtwebster** [comment]
This looks like an issue with viber, though I can't check as there's no source available. Viber is not providing any menu data.

Message when using StatusNotifier (Xapp status applet in cinnamon):
```
(xapp-sn-watcher:17660): LIBDBUSMENU-GLIB-WARNING **: 11:08:25.276: Getting layout failed: GDBus.Error:org.freedesktop.DBus.Error.UnknownObject: No such object path '/NO_DBUSMENU'
```
This shows an empty menu using the old system tray applet as well, and is broken in Mate.

**Comment by vushe** [comment]
Thanks @mtwebster for checking! Yes, seems to be an issue with Viber. Found a [workaround](https://askubuntu.com/questions/1434563/unable-to-quit-the-latest-version-of-viber). Perhaps, helps someone else.

**Comment by gocemitevski** [comment]
@mtwebster @vushe disabling the XApp Status Applet and restarting Cinnamon from the Troubleshooting menu, restores the right click menu in Viber. I'm using the Flatpak version. I still think this might have something to do with Cinnamon.

**Comment by mtwebster** [comment]
Disabling the XApp applet forces viber to use the legacy tray implementation (that *doesn't* use dbus).

The behavior is the same in xfce4 and MATE desktops - no menu.

I did make a small change to the xapp implementation that appears to allow viber to show a valid menu as long as its started after the 'xapp-sn-watcher' process (and after right-clicking once for an empty menu). At some point it still appears to lose it though.

There is still something going on in the app, however. I use quite a few Electron-based apps and have no issues whatsoever. Have you opened a bug report with Viber?

ref: https://github.com/linuxmint/xapp/commit/380fa81a1defccd5ab73e252bf9136cae992dc1d

**Comment by gocemitevski** [comment]
@mtwebster thanks! I found these similar issues in Gnome and KDE.

- https://bugs.kde.org/show_bug.cgi?id=449870
- https://askubuntu.com/questions/1434563/unable-to-quit-the-latest-version-of-viber

In terms of Electron-based apps, I have empty right click menu in Slack Desktop as well.

I'm happy to report a bug to Viber, though I'm not sure where Viber bugs on Linux should actually be reported.

**Comment by gocemitevski** [comment]
This appears to be fixed in Viber v18.2.0.2 that I'm currently running as a Flatpak app.
